File: java/src/main/java/com/cartenz/core/api/Rsa.java
package com.cartenz.core.api;

import android.content.Context;
import android.util.Base64;

import java.io.BufferedReader;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.security.KeyFactory;
import java.security.PublicKey;
import java.security.spec.X509EncodedKeySpec;
import java.util.ArrayList;
import java.util.List;

import javax.crypto.Cipher;

public class Rsa {
    public static String encryptRSA(Context ctx, String data, int resource) {
        try {
            InputStream is = ctx.getResources().openRawResource(resource);
            BufferedReader br = new BufferedReader(new InputStreamReader(is));
            List<String> lines = new ArrayList<String>();
            String line = null;
            while ((line = br.readLine()) != null)
                lines.add(line);

            // removes the first and last lines of the file (comments)
            if (lines.size() > 1 && lines.get(0).startsWith("-----")
                    && lines.get(lines.size() - 1).startsWith("-----")) {
                lines.remove(0);
                lines.remove(lines.size() - 1);
            }

            // concats the remaining lines to a single String
            StringBuilder sb = new StringBuilder();
            for (String aLine : lines)
                sb.append(aLine);
            String keyString = sb.toString();

            // converts the String to a PublicKey instance
            byte[] keyBytes = Base64.decode(keyString.getBytes("utf-8"),
                    Base64.DEFAULT);
            X509EncodedKeySpec spec = new X509EncodedKeySpec(keyBytes);
            KeyFactory keyFactory = KeyFactory.getInstance("RSA");
            PublicKey key = keyFactory.generatePublic(spec);

            // decrypts the message
            byte[] encryptedText = null;
            Cipher cipher = Cipher
                    .getInstance("RSA/ECB/PKCS1Padding");
            cipher.init(Cipher.ENCRYPT_MODE, key);
            encryptedText = cipher.doFinal(data.getBytes());
            return Base64.encodeToString(encryptedText, Base64.NO_WRAP);
        } catch (Exception e) {
            e.printStackTrace();
        }
        return null;
    }

}

File: kotlin/src/main/java/com/cartenz/kotlin_core/api/Rsa.kt
package com.cartenz.kotlin_core.api

import android.content.Context
import android.util.Base64
import java.io.BufferedReader
import java.io.InputStreamReader
import java.security.KeyFactory
import java.security.spec.X509EncodedKeySpec
import java.util.*
import javax.crypto.Cipher

public class Rsa() {
    companion object {
        fun encryptRSA(ctx: Context, data: String, resource: Int): String? {
            try {
                val `is` = ctx.resources.openRawResource(resource)
                val br = BufferedReader(InputStreamReader(`is`))
                val lines = ArrayList<String>()
                var line: String? = null
                while (line === br.readLine())
                    lines.add(line!!)

                // removes the first and last lines of the file (comments)
                if (lines.size > 1 && lines[0].startsWith("-----")
                        && lines[lines.size - 1].startsWith("-----")) {
                    lines.removeAt(0)
                    lines.removeAt(lines.size - 1)
                }

                // concats the remaining lines to a single String
                val sb = StringBuilder()
                for (aLine in lines)
                    sb.append(aLine)
                val keyString = sb.toString()

                // converts the String to a PublicKey instance
                val keyBytes = Base64.decode(keyString.toByteArray(charset("utf-8")),
                        Base64.DEFAULT)
                val spec = X509EncodedKeySpec(keyBytes)
                val keyFactory = KeyFactory.getInstance("RSA")
                val key = keyFactory.generatePublic(spec)

                // decrypts the message
                var encryptedText: ByteArray? = null
                val cipher = Cipher
                        .getInstance("RSA/ECB/PKCS1Padding")
                cipher.init(Cipher.ENCRYPT_MODE, key)
                encryptedText = cipher.doFinal(data.toByteArray())
                return Base64.encodeToString(encryptedText, Base64.NO_WRAP)
            } catch (e: Exception) {
                e.printStackTrace()
            }

            return null
        }

    }
}